kotlinx-coroutines-test/common/test/TestDispatchersTest.kt

```diff
@@ -4,16 +4,56 @@
 package kotlinx.coroutines.test
 
 import kotlinx.coroutines.*
+import kotlinx.coroutines.test.internal.*
 import kotlin.coroutines.*
 import kotlin.test.*
 
 class TestDispatchersTest: OrderedExecutionTestBase() {
 
     @BeforeTest
     fun setUp() {
+        Dispatchers.setMain(StandardTestDispatcher())
+    }
+
+    @AfterTest
+    fun tearDown() {
         Dispatchers.resetMain()
     }
 
+    /** Tests that asynchronous execution of tests does not happen concurrently with [AfterTest] (in fact, it does). */
+    @Ignore // fails on JS.
```

Comment:
For such kludges, we typically introduce helpers like [`noJs`](https://github.com/Kotlin/kotlinx.serialization/blob/5ce8e2cbd8d324ed5e0447f133137ccacc9f6614/formats/json/commonTest/src/kotlinx/serialization/test/TestHelpers.kt#L27)

Reply:
Good idea, using it now.